src/lib.rs

```diff
@@ -26,6 +26,7 @@
 )]
 #![allow(clippy::multiple_crate_versions, clippy::needless_doctest_main)]
 #![allow(bare_trait_objects)]
+#![recursion_limit = "1024"]
```

Comment:
What's going on here?

Reply:
the default value is 256 and `futures:stream` macro in the `read` method expands more than that so cargo clippy won't pass.